grabbit/core.py

```diff
@@ -573,3 +573,28 @@ def count_matches(f):
         matches = [m.path if return_type == 'file' else m.as_named_tuple()
                    for m in matches]
         return matches if all_ else matches[0] if matches else None
+
+
+def merge_layouts(layouts):
+    ''' Utility function for merging multiple layouts.for
+    Args:
+        layouts (list): A list of BIDSLayout instance to merge.
+    Returns:
+        A BIDSLayout containing merged files and entities.
+    Notes:
+        Layouts will be merged in the order of the elements in the list. I.e.,
+        the first Layout will be updated with all values in the 2nd Layout,
+        then the result will be updated with values from the 3rd Layout, etc.
+        This means that order matters: in the event of entity or filename
+        conflicts, later layouts will take precedence.
+    '''
+    for l in layouts[1:]:
+        layouts[0].files.update(l.files)
```

Comment:
Does it make sense to leave modify the inputs and instead do:

```Python
layout = layouts[0].clone()  # or copy()?

for l in layouts[1:]:
    layout.files.update(l.files)
    ...
```

I imagine most of the time people aren't going to merge and still want the original untouched, but the effect on the inputs is order-dependent, which feel wrong to me.